skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -0,0 +1,192 @@
+public class BtReceiverActivity extends InjectableActivity implements
+        BluetoothReceiver.Listener, BluetoothListAdapter.OnDeviceClickListener {
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.btn_refresh)
+    Button btnRefresh;
+
+    @BindView(R.id.list_bt_device)
+    RecyclerView recyclerView;
+
+    @Inject
+    ReceiverService receiverService;
+
+    @Inject
+    RxEventBus rxEventBus;
+
+    @Inject
+    BaseSchedulerProvider schedulerProvider;
+
+    private BluetoothReceiver bluetoothReceiver;
+    private final BluetoothListAdapter bluetoothListAdapter = new BluetoothListAdapter(this);
+    private boolean isConnected = false;
+    private ProgressDialog progressDialog;
+
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_bt_receive);
+        ButterKnife.bind(this);
+        initEvents();
+
+        setTitle(getString(R.string.connect_bluetooth_title));
+        setSupportActionBar(toolbar);
+
+        // checking for if bluetooth enabled
+        if (!BluetoothUtils.isBluetoothEnabled()) {
+            BluetoothUtils.enableBluetooth();
+            bluetoothListAdapter.rescan();
+        }
+    }
+
+    private void initEvents() {
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(),
+                linearLayoutManager.getOrientation());
+        recyclerView.addItemDecoration(dividerItemDecoration);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(bluetoothListAdapter);
+        bluetoothReceiver = new BluetoothReceiver(this, this);
+        BluetoothAdapter.getDefaultAdapter().startDiscovery();
+
+        // click to refresh the devices list.
+        btnRefresh.setOnClickListener((View v) -> {
+            if (BluetoothUtils.isBluetoothEnabled()) {
+                bluetoothListAdapter.rescan();
+            } else {
+                BluetoothUtils.enableBluetooth();
+                Toast.makeText(this, "bluetooth has been disabled, turning on...", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    /**
+     * Creates a subscription for listening to all hotspot events being send through the
+     * application's {@link RxEventBus}
+     */
+    private Disposable addBluetoothEventSubscription() {
+        return rxEventBus.register(BluetoothEvent.class)
+                .subscribeOn(schedulerProvider.io())
+                .observeOn(schedulerProvider.androidThread())
+                .subscribe(bluetoothEvent -> {
+                    switch (bluetoothEvent.getStatus()) {
+                        case CONNECTED:
+                            progressDialog.setMessage("Connected, start downloading...");
```

Comment:
Create string resource for this or use the same that is defined for hotspot.

Reply:
Fixed.